src/ripple/rpc/handlers/AccountTx.cpp

```diff
@@ -273,11 +276,283 @@ getLedgerRange(
     return LedgerRange{uLedgerMin, uLedgerMax};
 }
 
+std::variant<TxnsData, TxnsDataBinary>
+flatFetchTransactions(
+    RPC::Context& context,
+    std::vector<uint256>& nodestoreHashes,
+    std::vector<uint32_t>& ledgerSequences,
+    bool binary)
+{
+    std::variant<TxnsData, TxnsDataBinary> ret;
+    if (binary)
+        ret = TxnsDataBinary();
+    else
+        ret = TxnsData();
+    auto start = std::chrono::system_clock::now();
+    auto objs = context.app.getNodeFamily().db().fetchBatch(nodestoreHashes);
+
+    auto end = std::chrono::system_clock::now();
+    JLOG(context.j.debug()) << "account_tx Flat fetch time : "
+                            << ((end - start).count() / 1000000000.0);
+    assert(objs.size() == nodestoreHashes.size());
+    for (size_t i = 0; i < objs.size(); ++i)
+    {
+        auto& obj = objs[i];
+        auto& nodestoreHash = nodestoreHashes[i];
+        if (obj)
+        {
+            auto node = SHAMapAbstractNode::makeFromPrefix(
+                makeSlice(obj->getData()), SHAMapHash{nodestoreHash});
+            if (!node)
+            {
+                assert(false);
+                Throw<std::runtime_error>("Error making SHAMap node");
+            }
+            auto item = (static_cast<SHAMapTreeNode*>(node.get()))->peekItem();
+            if (item)
+            {
+                if (binary)
+                {
+                    auto& transactions = std::get<TxnsDataBinary>(ret);
+                    SerialIter it(item->slice());
+                    Blob txnBlob = it.getVL();
+                    Blob metaBlob = it.getVL();
+                    transactions.push_back(
+                        std::make_tuple(txnBlob, metaBlob, ledgerSequences[i]));
+                }
+                else
+                {
+                    auto& transactions = std::get<TxnsData>(ret);
+                    auto [txn, meta] = deserializeTxPlusMeta(*item);
+                    std::string reason;
+                    auto txnRet =
+                        std::make_shared<Transaction>(txn, reason, context.app);
```

Comment:
Have to call `setLedger()` on each `Transaction` object. Otherwise `date`, `inLedger` and `ledger_index` is missing from the response.